src/main/scala/chiseltest/backends/VPIInterface.scala

```diff
@@ -0,0 +1,427 @@
+package chiseltest.backends
+
+import java.io.File
+import java.nio.channels.FileChannel
+
+import chisel3._
+import firrtl.ir.{GroundType, IntWidth, Port}
+import logger.LazyLogging
+
+import scala.collection.immutable.ListMap
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.duration._
+import scala.concurrent.{blocking, Await, ExecutionContext, Future}
+import scala.language.implicitConversions
+import scala.sys.process.{Process, ProcessLogger}
+
+class VPIInterface(topPorts: Seq[Port], topName: String, commands: Seq[String])
+    extends SimulatorInterface
+    with LazyLogging {
+  def cycleCount: BigInt = _currentClock
+
+  def poke(signal: String, value: BigInt): Unit = {
```

Comment:
Should this be `override def`? (and similarly below)

Reply:
no need to override, in `SimulatorInterface`, it's an abstract def.